libkbfs/block_retrieval_queue.go

```diff
@@ -125,21 +131,27 @@ var _ BlockRetriever = (*blockRetrievalQueue)(nil)
 // newBlockRetrievalQueue creates a new block retrieval queue. The numWorkers
 // parameter determines how many workers can concurrently call Work (more than
 // numWorkers will block).
-func newBlockRetrievalQueue(numWorkers int,
+func newBlockRetrievalQueue(numWorkers int, numPrefetchWorkers int,
 	config blockRetrievalConfig) *blockRetrievalQueue {
 	q := &blockRetrievalQueue{
-		config:      config,
-		log:         config.MakeLogger(""),
-		ptrs:        make(map[blockPtrLookup]*blockRetrieval),
-		heap:        &blockRetrievalHeap{},
-		workerQueue: make(chan chan<- *blockRetrieval, numWorkers),
-		workers:     make([]*blockRetrievalWorker, 0, numWorkers),
-		doneCh:      make(chan struct{}),
+		config:           config,
+		log:              config.MakeLogger(""),
+		ptrs:             make(map[blockPtrLookup]*blockRetrieval),
+		heap:             &blockRetrievalHeap{},
+		workerCh:         make(chan struct{}, numWorkers),
```

Comment:
Good point about the request order.  But I still don't see why we'd rather have a thousand goroutines waiting to put on that channel, rather than a channel but a buffer of length 1000.  At the very least, it'll make debugging stack dumps easier.